zetaclient/orchestrator/chain_activate.go

```diff
@@ -0,0 +1,220 @@
+package orchestrator
+
+import (
+	"time"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+
+	btcobserver "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/observer"
+	btcrpc "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/rpc"
+	btcsigner "github.com/zeta-chain/zetacore/zetaclient/chains/bitcoin/signer"
+	evmobserver "github.com/zeta-chain/zetacore/zetaclient/chains/evm/observer"
+	evmsigner "github.com/zeta-chain/zetacore/zetaclient/chains/evm/signer"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	"github.com/zeta-chain/zetacore/zetaclient/common"
+	"github.com/zeta-chain/zetacore/zetaclient/config"
+)
+
+// WatchEnabledChains watches for run-time chain activation and deactivation
+func (oc *Orchestrator) WatchEnabledChains() {
+	oc.logger.Std.Info().Msg("WatchChainActivation started")
+
+	ticker := time.NewTicker(common.ZetaBlockTime)
+	for {
+		select {
+		case <-ticker.C:
+			oc.ActivateDeactivateChains()
+		case <-oc.stop:
+			oc.logger.Std.Info().Msg("WatchChainActivation stopped")
+			return
+		}
+	}
+}
+
+// ActivateDeactivateChains activates or deactivates chain observers and signers
```

Comment:
Is there a way to ad some coverage to this method?

Reply:
https://github.com/zeta-chain/node/pull/2411/commits/4f61c380bd8629ea044d7284eabf99c09b447131